### ISSUE TITLE
Alerting: false positive firing

### ISSUE THREAD

**Issue by dorinpirvu** [grafana/grafana#68942]
### 
# What went wrong?

**What happened**:

  - An alert fires on what appears to be a false positive

**What did you expect to happen**:

  - It should not fire

### 
# How do we reproduce it?

Step 1:
Prepare InfluxDB (v1.8) measurement containing:
- one analogic value,
- one discrete value (0/1, true/false)
- one analogic value as limit of the first
both are recorded every second.

Step 2:
Create alert:
A query, now-5m to now:
SELECT analogic_value FROM measurement WHERE $timeFilter and discrete_value=true
B query, now-5m to now:
SELECT analogic_limit FROM measurement WHERE $timeFilter
C expression:
REDUCE, function MIN, input A, mode STRICT (also tried with Drop non-numeric values)
D expression:
REDUCE, function LAST, input B, mode Drop non-numeric values
E expression:
MATH, $C >= $D, set as alert condition

Alert evaluation behavior:
evaluate every 2min, for 5min.
Configure no data and error handling:
NoData -> ok
Error -> Error

Basically, the alert should fire if discrete_value is true and the analogic value is higher than the limit in the last 5min (actually 10min = 5min sets the alert to pending, 5min to fire).

The alert is attached to a webhook that collects all similar alerts. When I check the details (i.e there is a dashboard that shows the evolution of the analogic and discrete values), at the time of firing (+/- 30min) the min value of the analogic is NOT higher than the limit for 10 or even 5 minutes. The discrete value is ON for less than a minute, during which time the analogic value is indeed over the limit.

### 
# What Grafana version are you using?

Grafana v9.5.2 (cfcea75916)

### 
### Is the bug inside a Dashboard Panel?

Not insinde a dashboard / panel

### Grafana Platform?

A downloaded binary

### User's OS?

Ubuntu 22.04.2 LTS (GNU/Linux 5.15.0-70-generic x86_64)

### User's Browser?

Microsoft Edge Version 112.0.1722.39 (Official build) (64-bit)

### Is this a Regression?

None

### Are Datasources involved?

InfluxDB 1.8

### Anything else to add?

I have defined around 175 similar alerts, only this one has this behavior of false positive.
Only for this combination of analogic and discrete values the discrete value is ON for less than 1 minute.
For other combinations, the alerts are firing correspondingly.

**Comment by dorinpirvu** [comment]
Hello everyone, anyone got any idea about this issue?
In the meantime I have defined several other alerts, of which one has the same behavior as mentioned.

Technical process backround:
Several air flowmeters are installed on air distributors for different valves. 
The air flow is correlated to the opening of the valves on the corresponding flowmeters.
Most valves are open for enough time to detect air leaks, but there are two valves (two alerts) that are open for less then 10s at a time, and the air flow is higher than a threshold (for the time the valve is open). Normally the alert shouldn't fire, but it does.

**Comment by cletter7** [comment]
Tagging @grafana/observability-metrics as owners of InfluxDB data source

**Comment by gtk-grafana** [comment]
I don't think this is caused by influx, it sounds like an issue with alerting where values that exceed the threshold for a duration shorter then the alert evaluation interval are still firing the alert. @gillesdemey what do you think?

**Comment by dorinpirvu** [comment]
hello, any news on this issue? thanks

**Comment by dorinpirvu** [comment]
up. this issue is still relevant, any updates? thanks

**Comment by dorinpirvu** [comment]
hello, any updates on this?
